src/AR/LocationUtils.tsx

```diff
@@ -0,0 +1,73 @@
+import { Alert, PermissionsAndroid, Platform } from 'react-native';
+import Geolocation, {
+  type GeoPosition,
+} from 'react-native-geolocation-service';
+import { magnetometer, type SensorData } from 'react-native-sensors';
+
+export type Location = {
+  latitude: number;
+  longitude: number;
+} | null;
```

Comment:
I would prefer that we did not define type as possibly null. Could we replace usages of `Location` with usages of `Location | null`